src/DvbData.cpp

```diff
@@ -488,8 +488,15 @@ bool Dvb::GetRecordings(ADDON_HANDLE handle)
     recinfo.iDuration     = recording.duration;
     recinfo.iGenreType    = recording.genre & 0xF0;
     recinfo.iGenreSubType = recording.genre & 0x0F;
-    recinfo.iChannelUid   = PVR_CHANNEL_INVALID_UID; // TODO: try searching by name
-    recinfo.channelType   = PVR_RECORDING_CHANNEL_TYPE_TV;
+    recinfo.iChannelUid   = PVR_CHANNEL_INVALID_UID;
+    recinfo.channelType   = PVR_RECORDING_CHANNEL_TYPE_UNKNOWN;
+
+    if (recording.channel)
+    {
+      recinfo.iChannelUid = recording.channel->id;
+      recinfo.channelType = (recording.channel->radio)
+          ? PVR_RECORDING_CHANNEL_TYPE_RADIO : PVR_RECORDING_CHANNEL_TYPE_TV;
```

Comment:
just for your info, because the commit msg only mentions api 5.0.0: iChannelUid was api 5.0.0, channelType was api 5.1.0